pkg/profile/artifact.go

```diff
@@ -37,19 +34,62 @@ func (p *Profile) CreateArtifacts(ctx context.Context) error {
 		if !ok {
 			return fmt.Errorf("object %v cannot be asserted to client.Object", o)
 		}
-		p.log.Info("creating...", "kind", obj.GetObjectKind().GroupVersionKind().Kind, "resource", obj.GetName())
-		if err := p.client.Create(ctx, obj); err != nil {
-			return fmt.Errorf("failed to create %s: %w", obj.GetObjectKind().GroupVersionKind().Kind, err)
+		if err := p.reconcileArtifact(obj); err != nil {
+			return err
 		}
 	}
 
 	p.log.Info("all artifacts created")
 	return nil
 }
+func (p *Profile) reconcileArtifact(desiredObj client.Object) error {
+	existingObj := desiredObj.DeepCopyObject().(client.Object)
```

Comment:
What's the point of the deepclone here if you are immediately getting it, and Object key is used from the desiredObj?

Reply:
I need an object of the correct type for resource get read into

Reply:
But you are converting it into client.Object.

Reply:
So, I don't know, maybe I'm getting something wrong here, but, if you just create a new client.Object and `Get` that, get will provide all the type and information that it further requires. It's not like, you update a field or something afterwards. So I don't think the DeepCopy in this case adds value? everything will be overwritten anyways?

Reply:
And and also, shouldn't you pass in `&existingObj` instead?

Reply:
Ignore me. It's an interface. 🤦